docs/architecture/stack/evm-state-manager/index.mdx

```diff
@@ -1,31 +1,249 @@
+## Merkle trees
 
-The relationship between accounts–which accounts control which ones, for example–is represented by a tree structure. The way a small twig is derived from a larger branch, and that from a trunk: in this way, the entire bifurcating, iterative history of the network state is retained.
+The Merkle tree and its variations are commonly used across EVM chains to store and retrieve 
+data about the state of every account on the blockchain. 
 
-linea-besu uses a particular version of this technology called a Sparse Merkle Tree: it uses default values to represent certain levels of branching in the tree–and if there has been no change to that default value, it means there has been no activity “further out the branch”, and therefore no need to store data regarding it. This allows the network to be much more efficient, at the level of data storage and other improvements based on that, than other implementations of Merkle trees.
+A Merkle tree is comprised of 'nodes' that branch off from each other. At the base is the 'root', 
+or state root, from which branches stem, and leaves stem from the branches. 
 
-The state manager in linea-besu is relatively simple, and has two main functions: updating the state of the network, and proof generation.
+<div class="img-large">
+  <div class="mermaid-medium">
+    ```mermaid
+        flowchart TD
+            A[root] --> B[node] & C[node]
+            B --> D[node] & E[node]
+            C --> F[node] & G[node]
+    ```
+  </div>
+</div>
 
-- The sequencer executes a block, and sends it not only to the trace generator for it to do its job, but also to the state manager. Upon receiving an executed block from the sequencer, the state manager updates the state, in the Merkle tree, of every account that was affected, as documented in the trace data.
-- The new values, represented by new hashes, are now the state of the network following that block.
-- This new network state is necessary to generate a proof for submission to Ethereum. Therefore, the state manager passes the information off to the prover.
+Each node, regardless of type, is represented by a cryptographic hash which encodes data about its 
+properties — for example, the contents of your account. Each hash encodes the hashes of its child 
+nodes. Taken to its full extent, this cascading system means the root encodes data of the state of 
+every single account on the blockchain. 
+
+Cryptographic hashes are deterministic, which means you can reverse the hash function to get the 
+data which it encoded. If you have the hash of the root—the only node without a parent—you can 
+theoretically derive from it the data of any node in the entire tree. 
+
+As a layer 2 (L2) network, Linea is in the business of making transacting faster and more efficient.
+Linea implements a sparse Merkle tree due to the inefficiencies of standard Merkle tree
```

Comment:
My takeaway from A's docs is that Linea uses SMT just for the proof gen/storage and that the standard PMT is actually used to maintain (Linea) world state as its more efficient for that purpose? I may have totally misinterpreted this??

Reply:
Yeah, this is not crystal clear to me either. I think we have it correct already though in that the SMT is a more efficient method for account storage because it means you can assume the contents of a node and its children without having to actually access their contents. I think it may confuse if we also state it is inefficient? Maybe it's something to do with Linea's fast block time, far faster than L1, that means the SMT can't be used for world state as well.